### PR TITLE
adding comment for __future__  module

### PR DESCRIPTION
**Pull request by Raarav**

### CODE REVIEW

**Comment by sylee957** [comment]
`__future__` is used by almost every sympy module, so there adding as a documentation under a single module is not much a viable option.

It would be better added to development tips [wiki](https://github.com/sympy/sympy/wiki#development), if there doesn't exists any section, and if it is intended to help new developers.

And I see most of the description is duplicate of [official python doc](https://docs.python.org/3.7/library/__future__.html), so a simple redirective can be fine.

File: sympy/tensor/index_methods.py [hunk 1]
```diff
@@ -12,6 +12,18 @@
 
 from __future__ import print_function, division
 
+ """
```
Comment (Sc0rpi0n101) [inline sympy/tensor/index_methods.py]:
```suggestion
"""
```
There's also an indentation issue here

File: sympy/tensor/index_methods.py [hunk 2]
```diff
@@ -12,6 +12,18 @@
...
+
+ __future__ is a real module, and serves three purposes:
+
+To avoid confusing existing tools that analyze import statements and expect to find the modules they�re importing.
+To ensure that future statements run under releases prior to 2.1 at least yield runtime exceptions (the import of __future__ will fail, because there was no module of that name prior to 2.1).
+To document when incompatible changes were introduced, and when they will be � or were � made mandatory. This is a form of executable documentation, and can be inspected programmatically via importing __future__ and examining its contents.
+Each statement in __future__.py is of the form:
+
+ """
```
Comment (Sc0rpi0n101) [inline sympy/tensor/index_methods.py]:
```suggestion
"""
```
and here

File: sympy/tensor/index_methods.py [hunk 3]
```diff
@@ -12,6 +12,18 @@
 
 from __future__ import print_function, division
 
+ """
+
+ __future__ is a real module, and serves three purposes:
+
+To avoid confusing existing tools that analyze import statements and expect to find the modules they�re importing.
+To ensure that future statements run under releases prior to 2.1 at least yield runtime exceptions (the import of __future__ will fail, because there was no module of that name prior to 2.1).
```
Comment (Sc0rpi0n101) [inline sympy/tensor/index_methods.py]: This line and the next line easily extends the 79-character limit.
Please be careful not to have too long lines. It doesn't follow the Python and SymPy styling guide
[PEP-8](https://www.python.org/dev/peps/pep-0008/)

**Review by Sc0rpi0n101 — COMMENTED** [review]
Yeah, sylee is right, redirection will be more effective as the official page will always be more descriptive than a docstring.
and it will lead to smaller docstrings which can be included in other modules too as it won't cause much inconvenience

**Comment by asmeurer** [comment]
@sylee957 is right that `__future__` does not need to be documented in the code.

Also, do not directly copy text from other sources without attribution. This is plagiarism.

**Comment by Raarav** [comment]
thanks to all give me right suggestions on right way i will considered it
and next time i will implement some valueable contribution to Sympy. thanks
to  @aaron @nikhil @S.Y. Lee

File: sympy/tensor/index_methods.py (hunk 3 above)
Comment (Raarav) [reply sympy/tensor/index_methods.py]: ok next time i will considered it

**Comment by oscargus** [comment]
Before reading the comments, I computed the number of times __future__ is mentioned in the source: 529.

**Comment by Sc0rpi0n101** [comment]
It's @Sc0rpi0n101 ,
My name is Nikhil Maan, not the alias, @nikhil is someone else.
It's no problem for me,
but don't @ some random dude xD

> thanks to all give me right suggestions on right way i will considered it and next time i will implement some valueable contribution to Sympy. thanks to @aaron @nikhil @S.Y. Lee

**Comment by Raarav** [comment]
sorry for that @Sc0rpi0n101 ,

**Comment by Sc0rpi0n101** [comment]
Don't worry 
I don't have a problem with that. :)
But just be careful tagging some guy who might not even know what this is about

**Comment by Raarav** [comment]
next time i will comment with right github username. thanks to correct me. @Sc0rpi0n101

**Comment by Raarav** [comment]
> Don't worry
> I don't have a problem with that. :)
> But just be careful tagging some guy who might not even know what this is about

**Comment by Raarav** [comment]
> Don't worry
> I don't have a problem with that. :)
> But just be careful tagging some guy who might not even know what this is about

ok

**Pull request closed without merging**